src/components/RevokeContainer/RevokeContainer.tsx

```diff
@@ -0,0 +1,110 @@
+import { FunctionComponent, useState, useEffect } from "react";
+import { Redirect } from "react-router";
+import { useConfigContext } from "../../common/context/config";
+import { RevokeDocumentDropZone } from "./RevokeDocumentDropZone/RevokeDocumentDropZone";
+import { RevokeDocumentTileArea } from "./RevokeDocumentTileArea/RevokeDocumentTileArea";
+import { utils } from "@govtechsg/open-attestation";
+import { RevokeConfirmationModal } from "./RevokeConfirmationModal";
+import { verificationBuilder, isValid, openAttestationVerifiers } from "@govtechsg/oa-verify";
+import { DocumentUploadState } from "../../constants/DocumentUploadState";
+import { ProcessDocumentScreen } from "../ProcessDocumentScreen";
+import { QueueType } from "../../constants/QueueState";
+
+export const RevokeContainer: FunctionComponent = () => {
+  const { config } = useConfigContext();
+  const [revokeDocuments, setRevokeDocuments] = useState([]);
+  const [fileName, setFileName] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [showConfirmationModal, setShowConfirmationModal] = useState(false);
+  const [revokeStep, setRevokeStep] = useState(1);
+  const [documentUploadState, setDocumentUploadState] = useState(DocumentUploadState.INITIALIZED);
+
```

Comment:
for complex state management like this component, its better to either:
1. use something like redux + sagas to handle it (i know we currently dont have anything like that in this repo)
2. use useReducer hook to at least make some of the functions more pure by dispatching actions in the sub functions.

then you can move certain pieces of code out of the component to keep it more manageable.
for now its fine, but its something to keep in mind.

Reply:
ok